geoportal/c2cgeoportal_geoportal/views/layers.py

```diff
@@ -256,52 +382,22 @@ def count(self) -> int:
 
     @view_config(route_name="layers_create", renderer="geojson")  # type: ignore
     def create(self) -> Optional[FeatureCollection]:
-        from c2cgeoportal_commons.models.main import (  # pylint: disable=import-outside-toplevel
-            Layer,
-            RestrictionArea,
-            Role,
-        )
-
         set_common_headers(self.request, "layers", Cache.PRIVATE_NO)
 
         if self.request.user is None:
             raise HTTPForbidden()
 
         self.request.response.cache_control.no_cache = True
 
-        layer = self._get_layer_for_request()
-
-        def check_geometry(_: Any, feature: Feature, obj: Any) -> None:
-            del obj  # unused
-            geom = feature.geometry
-            if geom and not isinstance(geom, geojson.geometry.Default):
-                shape = shapely.geometry.shape(geom)
-                srid = self._get_geom_col_info(layer)[1]
-                spatial_elt = from_shape(shape, srid=srid)
-                allowed = models.DBSession.query(func.count(RestrictionArea.id))
-                allowed = allowed.join(RestrictionArea.roles)
-                allowed = allowed.join(RestrictionArea.layers)
-                allowed = allowed.filter(RestrictionArea.readwrite.is_(True))
-                allowed = allowed.filter(Role.id.in_(get_roles_id(self.request)))
-                allowed = allowed.filter(Layer.id == layer.id)
-                allowed = allowed.filter(
-                    or_(RestrictionArea.area.is_(None), RestrictionArea.area.ST_Contains(spatial_elt))
-                )
-                if allowed.scalar() == 0:
-                    raise HTTPForbidden()
-
-                # Check if geometry is valid
-                if self._get_validation_setting(layer):
-                    self._validate_geometry(spatial_elt)
-
-        protocol = self._get_protocol_for_layer(layer, before_create=check_geometry)
+        protocol = self._get_protocol_for_request()
+        log_updater = _LogLastUpdate(self._get_layer_for_request())
         try:
             features = protocol.create(self.request)
             if isinstance(features, HTTPException):
                 raise features
             if features is not None:
                 for feature in features.features:  # pylint: disable=no-member
-                    self._log_last_update(layer, feature)
+                    log_updater.update(self.request, feature)
```

Comment:
Do not we still have an UPDATE request after then INSERT request here ?

Reply:
I look like to be true. I will do deeper checks, thanks :-)

Reply:
Yes, but to avoid that, we should add an after delete in papyrus!